File: store-java/src/main/java/io/mb/store/model/Item.java
package io.mb.store.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

import static io.mb.store.resource.Constants.*;

@Getter
@Setter
@Entity
@Table(schema = DB_SCHEMA, name = ITEM_TABLE)
public class Item implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = ITEM_SEQUENCE)
    @SequenceGenerator(name = ITEM_SEQUENCE)
    private Long id = null;

    @Column(unique = true, nullable = false)
    private String vin;

    @Column(nullable = false)
    private String dealerId;
}

File: store-kotlin/src/main/kotlin/io/mb/store/model/Item.kt
package io.mb.store.model

import io.mb.store.resource.DB_SCHEMA
import io.mb.store.resource.ITEM_SEQUENCE
import io.mb.store.resource.ITEM_TABLE
import jakarta.persistence.*

@Entity
@Table(schema = DB_SCHEMA, name = ITEM_TABLE)
data class Item(
    @Id
    @GeneratedValue(
        strategy = GenerationType.SEQUENCE,
        generator = ITEM_SEQUENCE
    )
    @SequenceGenerator(name = ITEM_SEQUENCE)
    val id: Long? = null,
    @Column(unique = true, nullable = false)
    val vin: String,
    @Column(nullable = false)
    val dealerId: String
)